src/vs/workbench/contrib/notebook/browser/viewParts/notebookViewZones.ts

```diff
@@ -29,25 +31,39 @@ export class NotebookViewZones extends Disposable {
 		this.domNode.setPosition('absolute');
 		this.domNode.setAttribute('role', 'presentation');
 		this.domNode.setAttribute('aria-hidden', 'true');
+		this.domNode.setWidth('100%');
 		this._zones = {};
 
 		this.listView.containerDomNode.appendChild(this.domNode.domNode);
 	}
 
-	changeViewZones(callback: (changeAccessor: INotebookViewZoneChangeAccessor) => void): void {
+	changeViewZones(callback: (changeAccessor: INotebookViewZoneChangeAccessor) => void): boolean {
+		let zonesHaveChanged = false;
 		const changeAccessor: INotebookViewZoneChangeAccessor = {
 			addZone: (zone: INotebookViewZone): string => {
+				zonesHaveChanged = true;
 				return this._addZone(zone);
 			},
 			removeZone: (id: string): void => {
+				zonesHaveChanged = true;
+				// TODO: validate if zones have changed layout
 				this._removeZone(id);
 			},
 			layoutZone: (id: string): void => {
+				zonesHaveChanged = true;
+				// TODO: validate if zones have changed layout
```

Comment:
Intentionally not included in PR?

Reply:
Right, this one needs more thinking, I plan to tackle this later.